animalcontrol.lua

```diff
@@ -0,0 +1,195 @@
+-- animalcontrol helps manage the butchery and gelding of animals
+-- Written by Josh Cooper(cppcooper) on 2020-02-18, last modified: 2020-02-19
+local utils=require('utils')
+local validArgs = utils.invert({
+ 'all',
+ 'id',
+ 'race',
+ 'markedfor',
+ 'notmarkedfor',
+ 'gelded',
+ 'notgelded',
+ 'male',
+ 'female',
+
+ 'showstats',
+ 'markfor',
+ 'unmarkfor',
+
+ 'help',
+})
+local args = utils.processArgs({...}, validArgs)
+local help = [====[
+
+animalcontrol
+=============
+Animal control is a script useful for deciding what animals to butcher and geld.
+
+I developed this because I found Dwarf Therapist had a bug for my version of
+dwarf fortress which prevented the geld status of animals from updating. So
+I would be queuing animals to be gelded that had already been gelded.
+
+While not as powerful as Dwarf Therapist in managing animals - in so far as
+DT allows you to sort by various stats and flags - this script does provide
+many options for filtering animals. Additionally you can mark animals for
+slaughter or gelding, you can even do so enmasse if you so choose.
+
+Examples:
+  [DFHack]# animalcontrol -race 170
```

Comment:
Most DFHack users have no idea how to obtain the numerical ID of a particular race, and this varies greatly when mods come into play. If this is intended to work as a standalone script, it might be better to allow users to specify a race by its raw-defined name. The following is a modified excerpt from modtools/create-unit.lua which exemplifies how this might be done:

```lua
local raceID
for i,c in ipairs(df.global.world.raws.creatures.all) do
  if c.creature_id == args.race then
    raceID = i
    break
  end
end
if not raceID then
  qerror('Invalid race: ' .. args.race)
end
```

Reply:
It lists all information required for further command narrowing. I can give a shot at adding that code and see if it functions as required